### PR TITLE
Im 73 users dont use email as unique and primary key 2

### PR DESCRIPTION
**Pull request by kristinaBc3**

### CODE REVIEW

File: klab.hub/src/main/java/org/integratedmodelling/klab/hub/api/TokenVerifyEmailClickback.java [hunk 1]
```diff
@@ -1,22 +1,40 @@
 package org.integratedmodelling.klab.hub.api;
 
+import org.integratedmodelling.klab.hub.config.LinkConfig;
 import org.springframework.data.annotation.TypeAlias;
 
 @TypeAlias("VerifyEmail")
 public class TokenVerifyEmailClickback extends TokenClickback {
 	
 	private static final long serialVersionUID = 2577854654763037014L;
+	private String notVerifiedEmail;
```
Comment (iperdomo) [inline klab.hub/src/main/java/org/integratedmodelling/klab/hub/api/TokenVerifyEmailClickback.java:10]: Question about naming (we know that naming is hard 😅): `notVerifiedEmail` vs `newEmail` ?

File: klab.hub/src/main/java/org/integratedmodelling/klab/hub/tokens/services/RegistrationTokenServiceImpl.java [hunk 2]
```diff
@@ -34,6 +35,17 @@ public TokenClickback createToken(String username, TokenType type) {
 			return new CreateChangePasswordToken(repository, username, linkConfig).execute();
 		} else if(type.equals(TokenType.lostPassword)) {
 			return new CreateLostPasswordToken(repository, username, linkConfig).execute();
+		} else if(type.equals(TokenType.verifyEmail)) {
+			return new CreateVerifyEmailToken(repository, username, linkConfig).execute();
+		} else {
+			return null;
```
Comment (iperdomo) [inline klab.hub/src/main/java/org/integratedmodelling/klab/hub/tokens/services/RegistrationTokenServiceImpl.java:41]: Do we want to return `null` or throw an Exception? 🤔

File: klab.hub/src/main/java/org/integratedmodelling/klab/hub/users/controllers/UserProfileController.java [hunk 3]
```diff
@@ -103,6 +120,23 @@ public ResponseEntity<?> getUserProfile(@PathVariable String id) {
...
+	@GetMapping(value=API.HUB.USER_BASE_ID_NOAUTH, params = API.HUB.PARAMETERS.USER_GET)
+	public ResponseEntity<?> getUserDataByToken(@PathVariable String id, @RequestParam(API.HUB.PARAMETERS.USER_GET) String setEmail) {
+		TokenVerifyEmailClickback token = null;
+		try {
+			token = (TokenVerifyEmailClickback) tokenService.getAndVerifyToken(id, setEmail, TokenType.verifyEmail);
+		} catch (Exception e) {
+			throw new ActivationTokenFailedException("User Verification token failed");
+		}
+		if (token == null) {
+			throw new ActivationTokenFailedException("User Verification token failed");
```
Comment (inigo-cobian) [inline klab.hub/src/main/java/org/integratedmodelling/klab/hub/users/controllers/UserProfileController.java:132]: Why don't we use a different error message for the missing token and the exception?

Comment (kristinaBc3) [reply klab.hub/src/main/java/org/integratedmodelling/klab/hub/users/controllers/UserProfileController.java:132]: Is unreacheable code, I'll delete it.

File: klab.hub/src/main/java/org/integratedmodelling/klab/hub/tokens/services/RegistrationTokenServiceImpl.java (hunk 2 above)
Comment (kristinaBc3) [reply klab.hub/src/main/java/org/integratedmodelling/klab/hub/tokens/services/RegistrationTokenServiceImpl.java:41]: Is for keeping the same structure.

**Pull request merged**